### ISSUE TITLE
Documentation

### ISSUE THREAD

**Issue by elibre82** [microsoft/qsharp#1144]
Hello,

I'm a professor in a university. I'm trying to build a course on quantum computing based on qsharp. I need the API documentation for this course. But it's nowhere now. There is no API documentation on the new "Modern QDK" site ! Such as this doc: https://learn.microsoft.com/en-us/qsharp/api/qsharp/microsoft.quantum.intrinsic.measure. This page still exists, but there is no more links to it...
Do you intend to include the doc ?

Thanks

**Comment by billti** [comment]
Working on it right now! Will have something up shortly and will update this issue when we do. Apologies for the delay.

**Comment by billti** [comment]
For now, it's all implemented int the files under https://github.com/microsoft/qsharp/tree/main/library/std, which are quite browsable. If you see anything you need, or have any other feedback in general on how this can be more useful for you, please do open issues with feedback and suggestions. Thanks!

**Comment by elibre82** [comment]
Hello,
Yes I saw this page on GitHub. What worried me is that it's not really user friendly and "searchable". For example, in an exercice I want to give to my students, they need "ControlledOnBitString" statement. I would have liked to give them the direct link to the page in the API. In the current site, it's not possible anymore, and searching the page is tedious because it can return a lot of results, like if you have to search for "X" gate in intrinsic library, or not one, like for "ControlledOnBitString" in canon library.
So a browsable doc, with a summary which groups the statements and a page per statement, like the previous one will be fantastic.
Thanks !

**Comment by elibre82** [comment]
Let me complete my previous answer. One of the strong arguments, among a lot of others, in favor of adopting the Q# language for my course in quantum computing was the quality of the documentation. The only thing missing from the previous documentation was the ability to do a global search in libraries. For example, if I want to search for "ControlledOnBitString" statement but I don't know which library it is in. 
My course is a little over a month away. I have already included almost all the links to the old API I need in my lab docs. Do you intend to migrate the doc by then?
Thanks

**Comment by billti** [comment]
Yes. We're working on building the API docs from the repo that we can push to the usual docs location on learn.microsoft.com. That work is in progress (see https://github.com/microsoft/qsharp/pull/1150), and hopefully will be live in a week or two.

**Comment by billti** [comment]
> I have already included almost all the links to the old API I need in my lab docs

If you want to share the links you had included, I can work with the docs team to ensure they are ones we intend to keep, or will provide you with updates for any that may change.

**Comment by elibre82** [comment]
Hello,

Here are some of the links I use. I I probably missed a few...
https://learn.microsoft.com/en-us/qsharp/api/qsharp/microsoft.quantum.diagnostics
https://learn.microsoft.com/en-us/qsharp/api/qsharp/microsoft.quantum.diagnostics.dumpmachinene
https://learn.microsoft.com/en-us/qsharp/api/qsharp/microsoft.quantum.intrinsic
https://learn.microsoft.com/en-us/qsharp/api/qsharp/microsoft.quantum.intrinsic.r
https://learn.microsoft.com/en-us/qsharp/api/qsharp/microsoft.quantum.math
https://learn.microsoft.com/en-us/qsharp/api/qsharp/microsoft.quantum.math.arctan2
https://learn.microsoft.com/en-us/qsharp/api/qsharp/microsoft.quantum.intrinsic.m
https://learn.microsoft.com/en-us/qsharp/api/qsharp/microsoft.quantum.intrinsic.measure
https://learn.microsoft.com/en-us/azure/quantum/user-guide/language/expressions/iterations
https://learn.microsoft.com/en-us/azure/quantum/user-guide/language/typesystem/immutability
https://learn.microsoft.com/en-us/qsharp/api/qsharp/microsoft.quantum.math.powi
https://learn.microsoft.com/en-us/qsharp/api/qsharp/microsoft.quantum.math.arccos
https://learn.microsoft.com/en-us/qsharp/api/qsharp/microsoft.quantum.math.sqrt
https://learn.microsoft.com/en-us/qsharp/api/qsharp/microsoft.quantum.math.bitsizei
https://learn.microsoft.com/en-us/qsharp/api/qsharp/microsoft.quantum.diagnostics.dumpoperation
https://learn.microsoft.com/en-us/qsharp/api/qsharp/microsoft.quantum.intrinsic.cnot
https://learn.microsoft.com/en-us/qsharp/api/qsharp/microsoft.quantum.intrinsic.swap
https://learn.microsoft.com/en-us/azure/quantum/user-guide/language/expressions/functorapplication#controlled-functor
https://learn.microsoft.com/en-us/qsharp/api/qsharp/microsoft.quantum.canon.controlledonbitstring
https://learn.microsoft.com/en-us/qsharp/api/iqsharp-magic/trace
https://learn.microsoft.com/en-us/qsharp/api/iqsharp-magic/debug
https://learn.microsoft.com/en-us/azure/quantum/user-guide/language/expressions/conditionalbranching
https://learn.microsoft.com/en-us/qsharp/api/qsharp/microsoft.quantum.canon.controlledonint
https://learn.microsoft.com/en-us/azure/quantum/user-guide/language/expressions/conjugations

Thanks !

**Comment by elibre82** [comment]
I forgot this one:
https://learn.microsoft.com/en-us/qsharp/api/qsharp/microsoft.quantum.arithmetic

**Comment by billti** [comment]
Hi @elibre82 

We've just pushed the API docs for the new 'Modern QDK'. For example, you can find the new docs for some of the APIs you link to above at:

https://learn.microsoft.com/en-us/qsharp/api/qsharp-lang/microsoft.quantum.diagnostics/dumpmachine
https://learn.microsoft.com/en-us/qsharp/api/qsharp-lang/microsoft.quantum.intrinsic/r
etc.

(You should be able to find the other APIs easy enough from the left-hand side navigation tree)

Those docs under /quantum/user-guide will continue to exist.

The iqsharp magic commands no longer exist in the Modern QDK. You can debug through a Q# file now in the VS Code extension as for many other programming languages (see https://learn.microsoft.com/en-us/azure/quantum/user-guide/testing-debugging).

There are a couple APIs you'll find no longer exist in the modern QDK, as we didn't port some we felt weren't necessary. We are open to adding some back if there is need (e.g. we're currently adding back DumpRegister - see https://github.com/microsoft/qsharp/pull/1173 ). If there are specific APIs you don't see in the Modern QDK now but would like back, please do open a new issue with the API name and the reason for needing it.

**Comment by elibre82** [comment]
Hi @billti,
Thank ! I'll have a look on all this.
